src/WebUI/ImageDetails/ImageDetailsStore.ts

```diff
@@ -33,13 +35,28 @@ export class ImageDetailsStore extends StoreBase {
         return undefined;
     }
 
+    public getImageDetails(imageName: string): IImageDetails | undefined {
+        if (this._imageDetails && this._imageDetails.hasOwnProperty(imageName)) {
```

Comment:
we can simply say

return this._imageDetails[imageName];

reason is this._imageDetails is never null and this._imageDetails[imageName] can return null or undefined if does not exist, if exists will send the actual data, that actual data could be null also.

we can make similar change to hasImageDetails

Reply:
Done